internal/driver/fetch.go

```diff
@@ -492,7 +492,9 @@ mapping:
 func fetch(source string, duration, timeout time.Duration, ui plugin.UI, tr http.RoundTripper) (p *profile.Profile, src string, err error) {
 	var f io.ReadCloser
 
-	if sourceURL, timeout := adjustURL(source, duration, timeout); sourceURL != "" {
+	if isFile(source) {
```

Comment:
I think this code as written breaks opening perf.data files? We never get to that condition now.

Reply:
I think it shouldn't be the case. The previous logical sequence is to first judge adjustURL. If the conditions are not met, then judge isPerfFile. If the conditions are still not met, os.Open is executed at last. The changed logic is just to determine whether the source string corresponds to a file in the current directory before executing adjustURL. If so, it will be treated as a file first regardless of whether it contains a colon.  

I think it is equivalent to put the isFile(source) judgment in the adjustURL func, just like I did when I submitted it for the first time.

Reply:
I'm talking just about the current code, for simplicity. Here is what I think happens now: We first check `isFile()` which will return true for a perf.data file and so we will never get to the `isPerfFile()` check or `convertPerfData()` call. Instead we will try to parse the perf.data file content as a profile.proto which will of course fail. Does this make sense?

Reply:
This problem does exist. Previously, isPerfFile and convertPerfData were executed for local files in perf format that did not contain colons. But now it will go to isFile.

I should adjust the order of execution and put the isPerfFile check first if isFile is true, and execute adjustURL if it is false.